thrall/app/lib/elasticsearch/GoToGoCheck.scala

```diff
@@ -0,0 +1,63 @@
+package lib.elasticsearch
+
+import com.amazonaws.util.EC2MetadataUtils
+import com.gu.mediaservice.lib.elasticsearch.MappingTest
+import com.gu.mediaservice.lib.logging.LogMarker
+import com.sksamuel.elastic4s.Response
+import com.sksamuel.elastic4s.requests.delete.DeleteResponse
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * This implements a good to go check that validates when Thrall starts up that it can successfully write and read
+  * and image into the elasticsearch node.
+  */
+object GoToGoCheck extends StrictLogging {
+  def run(es: ElasticSearch)(implicit ex: ExecutionContext): Future[Unit] = {
+    // make a test image with an ID that is unique to the node so that multiple nodes can run the check at once
+    val uniqueId = Option(EC2MetadataUtils.getInstanceId).map(_.stripPrefix("i-")).getOrElse("abcdef")
+    val image = MappingTest.testImage.copy(id = s"${MappingTest.testImage.id}$uniqueId")
+    val lastModified = image.lastModified.get
+    implicit val marker: LogMarker = image ++ Map("goToGo" -> true)
+
+    val indexResult: Future[Unit] =
+      for {
+        // perhaps a test image wasn't previously cleaned up, let's err on the side of caution and check and deal with it
+        _ <- Future.successful(logger.info(s"Ensuring test image ${image.id} doesn't exist"))
```

Comment:
Vaguely concerned about races here.  Either they have unique ids and we don't need to do this, or they don't and we have a possibility of two new thralls fighting.  Unless this is more oriented to local testing?  If so, perhaps a comment to that effect?

Reply:
I did indeed hit this when doing local testing and at that point introduced the deletion and shortly after also aded unique IDs so that there were not races between nodes. You're quite right that this won't do a lot - unless thrall is restarting in a loop on the same node??? I will do some more thinking.

Reply:
I've added a bit of extra context here to describe when this might happen.